pkg/network/self_sign_configuration_test.go

```diff
@@ -0,0 +1,354 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Testing SelfSignConfiguration", func() {
+	Describe("validation function", func() {
+		Context("When CARotateInterval is the only empty one", func() {
+			It("should return an error", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					SelfSignConfiguration: &opv1alpha1.SelfSignConfiguration{
+						CARotateInterval:   "",
+						CAOverlapInterval:  "48h30m50s",
+						CertRotateInterval: "48h30m50s",
+					},
+				}
+				errorList := validateSelfSignConfiguration(clusterConfig)
+				Expect(errorList).To(HaveLen(1), "validation failed due to an unexpected error: %v", errorList)
+				Expect(errorList[0].Error()).To(Equal("failed to validate selfSignConfiguration: caRotateInterval is missing"))
+			})
+		})
+		Context("When CAOverlapInterval is the only empty one", func() {
+			It("should return an error", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					SelfSignConfiguration: &opv1alpha1.SelfSignConfiguration{
+						CARotateInterval:   "48h30m50s",
+						CAOverlapInterval:  "",
+						CertRotateInterval: "48h30m50s",
+					},
+				}
+				errorList := validateSelfSignConfiguration(clusterConfig)
+				Expect(errorList).To(HaveLen(1), "validation failed due to an unexpected error: %v", errorList)
+				Expect(errorList[0].Error()).To(Equal("failed to validate selfSignConfiguration: caOverlapInterval is missing"))
+			})
+		})
+		Context("When CertRotateInterval is the only empty one", func() {
+			It("should return an error", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					SelfSignConfiguration: &opv1alpha1.SelfSignConfiguration{
+						CARotateInterval:   "48h30m50s",
+						CAOverlapInterval:  "48h30m50s",
+						CertRotateInterval: "",
+					},
+				}
+				errorList := validateSelfSignConfiguration(clusterConfig)
+				Expect(errorList).To(HaveLen(1), "validation failed due to an unexpected error: %v", errorList)
+				Expect(errorList[0].Error()).To(Equal("failed to validate selfSignConfiguration: certRotateInterval is missing"))
+			})
+		})
```

Comment:
This can be written as a table checking that if one is missing, it return an error.